dbcon/joblist/tupleaggregatestep.cpp

```diff
@@ -2754,35 +2788,71 @@ void TupleAggregateStep::prep1PhaseDistinctAggregate(
             scaleAggSub.push_back(scaleProj[j]);
             precisionAggSub.push_back(precisionProj[j]);
             typeAggSub.push_back(typeProj[j]);
-            csNumAggSub.push_back(csNumProj[j]);
+            csNumAggSub.push_back(csNumProj[i]);
             widthAggSub.push_back(widthProj[j]);
 
-            // construct sub-rowgroup
-            posAggSub.clear();
-            posAggSub.push_back(2);   // rid
-
-            for (uint64_t k = 0; k < oidsAggSub.size(); k++)
-                posAggSub.push_back(posAggSub[k] + widthAggSub[k]);
-
-            RowGroup subRg(oidsAggSub.size(), posAggSub, oidsAggSub, keysAggSub, typeAggSub,
-                           csNumAggSub, scaleAggSub, precisionAggSub, jobInfo.stringTableThreshold);
-            subRgVec.push_back(subRg);
-
             // construct groupby vector
             vector<SP_ROWAGG_GRPBY_t> groupBySub;
-            uint64_t k = 0;
+            k = 0;
 
             while (k < jobInfo.groupByColVec.size())
             {
                 SP_ROWAGG_GRPBY_t groupby(new RowAggGroupByCol(k, k));
                 groupBySub.push_back(groupby);
                 k++;
             }
-
             // add the distinct column as groupby
             SP_ROWAGG_GRPBY_t groupby(new RowAggGroupByCol(j, k));
             groupBySub.push_back(groupby);
 
+           // Add multi parm distinct
+            while ((i+1) < returnedColVec.size() && functionIdMap(returnedColVec[i+1].second) == ROWAGG_MULTI_PARM)
+            {
+                ++i;
+                uint32_t dColKey = -1;
+                // Find the entry in distinctColVec, if any
+                for (k = 0; k < jobInfo.distinctColVec.size(); k++)
+                {
+                    dColKey = jobInfo.distinctColVec[k];
+                    if (returnedColVec[i].first  == dColKey)
+                        break;
+                    
+                }
+                idbassert(dColKey != (uint32_t)-1);
+                // locate the distinct key in the row group
+                for (k = 0; k < keysAgg.size(); k++)
+                {
+                    if (keysProj[k] == dColKey)
+                    {
+                        j = k;
+                        break;
+                    }
+                }
+                idbassert(j != -1);
```

Comment:
Looks like `j` should be set to `-1` somewhere at the begging of the `while` loop.

Reply:
Fixed